### PR TITLE
New functions for NPC

### PR DESCRIPTION
**Pull request by Supertos**
Added setSchedule, hasSchedule, setState and getState functions.

### CODE REVIEW

**Comment by thegrb93** [comment]
E2 supports enums right? Don't need to use strings

**Comment by bigdogmat** [comment]
`E2Lib.registerConstant`, also don't use C style comments.

**Comment by Supertos** [comment]
I think it's better not to have too many constants just use strings as in npcDisp and npcRelationship functions.

**Comment by Supertos** [comment]
Also added way to give npc other weapons than HL2 weapons.

**Comment by thegrb93** [comment]
Can you move your string lookup table creation outside of their respective functions? Recreating the table everytime the function is called is wasteful to cpu.

**Comment by Supertos** [comment]
Moved my "string lookup table creation" out of functions

File: lua/entities/gmod_wire_expression2/core/npc.lua [hunk 1]
```diff
@@ -152,3 +266,47 @@ e2function string entity:npcDisp(entity rv2)
...
+
+local function stateToStr(st)
+	local st2 = string.upper(st)
+
+	return STS[str2]
+end
+
+e2function void entity:setSchedule(string schedule)
+	if not IsValid(this) and this:IsNPC() then return end
+	if not strToSch(schedule)>=0 or not strToSch(schedule)<88 then return end
```
Comment (thegrb93) [inline lua/entities/gmod_wire_expression2/core/npc.lua]: This should just check for nil. Enum values are subject to change so these bounds are not reliable.

Comment (thegrb93) [reply lua/entities/gmod_wire_expression2/core/npc.lua]: You should also only call the strToSch function once and store its result as a local.

**Pull request closed without merging**